Remote JMX SSL bug fix and updated tests

roberttoyonaga opened the pull request:
This is a follow up to this pull request here: https://github.com/oracle/graal/pull/6250
 
It adds a reflection configuration that was missing, but needed for a JMX client to connect using SSL. The tests have been updated to connect with SSL and password authentication.

I've included dummy password, access, and SSL files required to configure the JMX properties needed for authentication and SSL.

roberttoyonaga commented:
Hi @fniephaus here's the fix I was mentioning and also the corresponding test updates.

fniephaus commented on substratevm/src/com.oracle.svm.test/src/com/oracle/svm/test/jmx/jmxremoteresources/servertruststore:
Is there anything important in here that needs to be documented?

roberttoyonaga replied:
I've now added some more explanation for why the ssl files are needed and how they were created [here](https://github.com/oracle/graal/pull/6275/files#diff-ecbd404c49c881d6e271dc72283d904a58891d47f64472a1c9829eb01f8e7dafR105-R109).  I also added the client.cer file (although it isn't really needed) for completeness.

fniephaus commented on substratevm/src/com.oracle.svm.test/src/com/oracle/svm/test/jmx/JmxTest.java:
```diff
@@ -89,7 +125,10 @@ private static MBeanServerConnection getLocalMBeanServerConnectionStatic() {
+            String[] credentials = {"myrole", "MYP@SSWORD"}; // dummy password for testing
+            env.put(JMXConnector.CREDENTIALS, credentials);
+            // Include below if protecting registry with SSL
+            env.put("com.sun.jndi.rmi.factory.socket", new SslRMIClientSocketFactory());
```
Should this key be a constant similar to the rest?

roberttoyonaga replied:
Good point. I've changed it now.

fniephaus reviewed (commented):
Thanks for the follow up PR. I left two minor comments.

fniephaus commented on substratevm/src/com.oracle.svm.test/src/com/oracle/svm/test/jmx/jmxremoteresources/client.cer:
`client.cer` currently uses the following information: `Issuer: C = ZZ, ST = GraalVM, L = GraalVM, O = GraalVM, OU = GraalVM, CN = GraalVM`.

I think we should avoid `GraalVM` here and use something like `JmxTest` instead.

roberttoyonaga replied:
ok I've updated it to use "JmxTest" now :)

roberttoyonaga commented:
> I went over the PR and added a commit (see #11599).
> 
> An internal reviewer mentioned we need to get rid of the binary files. Is there any way you could build on [19c02bd](https://github.com/oracle/graal/commit/19c02bdfc26d83fcb426a03feddb5b05f91c907a) and make is so that the keytool is invoked in the temporary directory?

Hi @fniephaus! Thank you for doing that. I've cherry-picked your commit and added a new commit on top of it here. The new commit uses keytool programmatically and creates the ssl files in the temporary directory.

fniephaus commented:
Thanks for taking care of this, @roberttoyonaga. I have refactored the test some more and now, all files are inlined into the test file and on-the-fly created in the temp directory.

if you could reset your branch to [this commit](https://github.com/oracle/graal/pull/11599/commits/be7138bdfe4dca53e63aea3cbc031e9bcd1d1e4a) and squash all commits, I think this is ready to be integrated.

fniephaus reviewed (approved):
LGTM, thank you @roberttoyonaga!
